Mlem/Logic/MlemLogger.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  MlemLogger.swift
+//  Mlem
+//
+//  Created by Sumeet Gill on 2024-01-31.
+//
+
+import Foundation
+import OSLog
+
+class MlemLogger {
+    private let logger: Logger
+    
+    required init(file: String = #fileID) {
+        self.logger = Logger(subsystem: AppConstants.loggerSubSystem, category: file)
```

Comment:
I made the category be the file name for now. Not sure if we should have defined categories for logging, such as "user interaction", "networking", etc. It may be a lot to maintain and enforce going forward if we do that.